src/Cli/dotnet/Parser.cs

```diff
@@ -118,6 +125,45 @@ public static Command GetBuiltInCommand(string commandName)
             .EnableLegacyDoubleDashBehavior()
             .Build();
 
+        private static async Task UseParseErrorReporting(string commandName, InvocationContext context, Func<InvocationContext, Task> next) 
+        {
+            CommandResult currentCommandResult = context.ParseResult.CommandResult;
+            while (currentCommandResult != null && currentCommandResult.Command.Name != commandName)
+            {
+                currentCommandResult = currentCommandResult.Parent as CommandResult;
+            }
+
+            if (currentCommandResult == null)
+            {
+                //different command was launched.
+                await next(context);
+                return;
+            }
+
+            if (context.ParseResult.Errors.Count > 0)
+            {
+                context.ExitCode = 127; //parse error
```

Comment:
@baronfel it would be good to set `InvocationResult` to `ParseErrorResult` at this point, let's aim to fix it after your fix introducing `HelpResult`.